sessionctx/variable/session.go

```diff
@@ -1372,97 +1372,6 @@ func (s *SessionVars) ClearStmtVars() {
 // i.e. oN / on / 1 => ON
 func (s *SessionVars) SetSystemVar(name string, val string) error {
 	switch name {
-	case TxnIsolationOneShot:
-		switch val {
-		case "SERIALIZABLE", "READ-UNCOMMITTED":
-			skipIsolationLevelCheck, err := GetSessionSystemVar(s, TiDBSkipIsolationLevelCheck)
-			returnErr := ErrUnsupportedIsolationLevel.GenWithStackByArgs(val)
-			if err != nil {
-				returnErr = err
-			}
-			if !TiDBOptOn(skipIsolationLevelCheck) || err != nil {
-				return returnErr
-			}
-			// SET TRANSACTION ISOLATION LEVEL will affect two internal variables:
-			// 1. tx_isolation
-			// 2. transaction_isolation
-			// The following if condition is used to deduplicate two same warnings.
-			if name == "transaction_isolation" {
-				s.StmtCtx.AppendWarning(returnErr)
-			}
```

Comment:
Why remove this part?

Reply:
I think that is an always-false branch. I've blamed the git history, but found that it is still an always-false condition. It looks like a copy&paste mistake. Check [this](https://github.com/pingcap/tidb/commit/28e62940daf1816a98d9aaca3fb6bf3b40f4759f).

Reply:
It's a switch based on `name` and this is under case `TxnIsolationOneShot`. So the branch for `name == "transaction_isolation"` is always false.